Src/Newtonsoft.Json/Serialization/JsonSerializerInternalReader.cs

```diff
@@ -1913,9 +1914,20 @@ private object PopulateObject(object newObject, JsonReader reader, JsonObjectCon
             OnDeserializing(reader, contract, newObject);
 
             // only need to keep a track of properies presence if they are required or a value should be defaulted if missing
-            Dictionary<JsonProperty, PropertyPresence> propertiesPresence = (contract.HasRequiredOrDefaultValueProperties || HasFlag(Serializer._defaultValueHandling, DefaultValueHandling.Populate))
-                ? contract.Properties.ToDictionary(m => m, m => PropertyPresence.None)
-                : null;
+            Dictionary<JsonProperty, PropertyPresence> propertiesPresence;
+            if (contract.HasRequiredOrDefaultValueProperties || HasFlag(Serializer._defaultValueHandling, DefaultValueHandling.Populate))
+            {
+                propertiesPresence = contract.Properties.ToDictionary(m => m, m => PropertyPresence.None);
+            }
+            else
+            {
+                if (contract.CreatedType.Name == "Root")
+                {
+                    Console.WriteLine("HERE on " + Thread.CurrentThread.ManagedThreadId);
```

Comment:
Why Console.WriteLine here?  Is there any logging or can you use builtin System.Diagnostics.Trace instead?